### PR TITLE
Add pdf support to utlegg

### PR DESCRIPTION
**Pull request by hhtjo**

### CODE REVIEW

File: src/AppBundle/Service/FileUploader.php [hunk 1]
```diff
@@ -84,7 +84,7 @@ public function uploadReceipt(Request $request)
 
         $mimeType = $file->getMimeType();
         $fileType = explode('/', $mimeType)[0];
-        if ($fileType === 'image') {
+        if ($fileType === 'image' || $mimeType === "application/pdf") {
             return $this->uploadFile($file, $this->receiptFolder);
         } else {
             throw new BadRequestHttpException('Filtypen må være et bilde.');
```
Comment (Lundli) [inline src/AppBundle/Service/FileUploader.php:90]:
```suggestion
            throw new BadRequestHttpException('Filtypen må være et bilde/pdf.');
```

File: app/Resources/views/receipt/create_receipt.html.twig [hunk 2]
```diff
@@ -21,9 +21,13 @@
 
         <div class="col-12 col-sm-6 d-flex flex-column">
             <div class="{{ receipt.picturePath is not null ? '' : 'd-none' }}">
-                <img class="img-fluid" src="{{ asset(receipt.picturePath) }}" id="receipt_picture">
+				<img class="img-fluid d-none" src="{{ asset(receipt.picturePath) }}" id="receipt_picture">
```
Comment (Lundli) [inline app/Resources/views/receipt/create_receipt.html.twig:24]: Indent seems a bit off here. :dart:

File: app/Resources/views/receipt/create_receipt.html.twig [hunk 3]
```diff
@@ -21,9 +21,13 @@
...
+				<object id="receipt_pdf" data="{{asset(receipt.picturePath)}}" class="d-none" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+					<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+						<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
+					</embed>
+				</object>
```
Comment (hhtjo) [inline app/Resources/views/receipt/create_receipt.html.twig:29]: Fix indent

File: app/Resources/views/receipt/create_receipt.html.twig [hunk 4]
```diff
@@ -21,9 +21,13 @@
 
         <div class="col-12 col-sm-6 d-flex flex-column">
             <div class="{{ receipt.picturePath is not null ? '' : 'd-none' }}">
-                <img class="img-fluid" src="{{ asset(receipt.picturePath) }}" id="receipt_picture">
+				<img class="img-fluid d-none" src="{{ asset(receipt.picturePath) }}" id="receipt_picture">
+				<object id="receipt_pdf" data="{{asset(receipt.picturePath)}}" class="d-none" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+					<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+						<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
```
Comment (Lundli) [inline app/Resources/views/receipt/create_receipt.html.twig:27]: All other text shown to the user is in Norwegian. 
The feedback should be as well for consistency.

File: app/Resources/views/receipt/edit_receipt.html.twig [hunk 5]
```diff
@@ -55,14 +55,35 @@
...
+									<object data="{{asset(receipt.picturePath)}}" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+										<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+											<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
+										</embed>
+									</object>
+									<br>
+									<a href="{{asset(receipt.picturePath)}}" target="_blank">Åpne PDF i ny fane</a>
+								</div>
+								<img class="img-fluid" src="{{ asset_with_version(receipt.picturePath) }}" id="receipt_picture">
+							{% endif %}
```
Comment (hhtjo) [inline app/Resources/views/receipt/edit_receipt.html.twig:80]: Fix indent

File: app/Resources/views/receipt/edit_receipt.html.twig [hunk 6]
```diff
@@ -77,10 +98,24 @@
...
+					  $('#receipt_pdf').find('a').attr('href', e.target.result);
+
+					  $('#receipt_picture').addClass('d-none');
+
+				  } else {
+					$('#receipt_pdf').addClass('d-none');
+
+					$('#receipt_picture').removeClass('d-none');
+					$('#receipt_picture').attr('src', e.target.result);
+				  }
```
Comment (hhtjo) [inline app/Resources/views/receipt/edit_receipt.html.twig:114]: Fix indent

File: app/Resources/views/receipt/my_receipts.html.twig [hunk 7]
```diff
@@ -88,7 +88,20 @@
...
+				  $('#receipt_pdf').find('a').attr('href', e.target.result);
+
+			  	  $('#receipt_picture').addClass('d-none');
+
+			  } else {
+			  	$('#receipt_pdf').addClass('d-none');
+
+			  	$('#receipt_picture').removeClass('d-none');
+				$('#receipt_picture').attr('src', e.target.result);
+			  }
```
Comment (hhtjo) [inline app/Resources/views/receipt/my_receipts.html.twig:104]: Fix indent

File: app/Resources/views/receipt/my_receipts_table.html.twig [hunk 8]
```diff
@@ -29,7 +29,17 @@
...
+										<object data="{{asset(receipt.picturePath)}}" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+											<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+												<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
+											</embed>
+										</object>
+										<br>
+										<a href="{{asset(receipt.picturePath)}}" target="_blank">Åpne PDF i ny fane</a>
+										{% else %}
+											<img class="img-fluid" src="{{ asset_with_version(receipt.picturePath) }}">
+										{% endif %}
```
Comment (hhtjo) [inline app/Resources/views/receipt/my_receipts_table.html.twig:42]: Fix indent

File: app/Resources/views/receipt/edit_receipt.html.twig [hunk 9]
```diff
@@ -55,14 +55,35 @@
                     </div>
                     <div class="col-4">
                         <div>
-                            <img src="{{ asset(receipt.picturePath) }}" class="img-fluid" id="receipt_picture">
+							{% if (receipt.picturePath matches '/\.pdf$/') %}
+								<div id="receipt_pdf">
+									<object data="{{asset(receipt.picturePath)}}" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+										<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+											<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
```
Comment (Lundli) [inline app/Resources/views/receipt/edit_receipt.html.twig:62]: ^Same as above

File: app/Resources/views/receipt_admin/receipts_table_individual.html.twig [hunk 10]
```diff
@@ -60,10 +60,18 @@
...
+									<object data="{{asset(receipt.picturePath)}}" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+										<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+											<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
+										</embed>
+									</object>
+									<br>
+									<a href="{{asset(receipt.picturePath)}}" target="_blank">Åpne PDF i ny fane</a>
+								{% else %}
+									<img class="img-fluid" src="{{ asset_with_version(receipt.picturePath) }}">
+								{% endif %}
```
Comment (hhtjo) [inline app/Resources/views/receipt_admin/receipts_table_individual.html.twig:74]: Fix indent

**Review by hhtjo — COMMENTED** [review]
Indentation needs to be fixed

File: app/Resources/views/receipt_admin/receipts_table_individual.html.twig [hunk 11]
```diff
@@ -60,10 +60,18 @@
...
                             </div>
-                            <div class="modal-body">
-                                <a href="{{ receipt.picturePath }}" target="_blank">
-                                    <img class="img-fluid" src="{{ receipt.picturePath }}">
-                                </a>
+                            <div class="modal-body mx-auto">
+								{% if (receipt.picturePath matches '/\.pdf$/') %}
+									<object data="{{asset(receipt.picturePath)}}" style="max-width: 100%; max-height:70%;" type="application/pdf" width="400" height="600">
+										<embed src="{{asset(receipt.picturePath)}}" type="application/pdf">
+											<p>This browser does not support PDFs. Please download the PDF to view it: <a href="{{asset(receipt.picturePath)}}">Download PDF</a>.</p>
```
Comment (Lundli) [inline app/Resources/views/receipt_admin/receipts_table_individual.html.twig:67]: Language.

**Review by Lundli — COMMENTED** [review]
Haven't tested the PR locally yet, just had a quick look in browser.

Looks good! :smile:  Had a few small remarks :)

**Review by Lundli — APPROVED** [review]

**Pull request merged**